testing/test_collection.py

```diff
@@ -244,28 +244,32 @@ def test_testpaths_ini(self, pytester: Pytester, monkeypatch: MonkeyPatch) -> No
         pytester.makeini(
             """
             [pytest]
-            testpaths = gui uts
+            testpaths = */tests
```

Comment:
I believe this is a backwards-incompatible change (and thus a problem) if the `testpaths` with `--pyargs` doesn't work like before anymore. I don't think this test still tests what it was supposed to (namely, that `testpaths` are Python modules rather than filesystem paths with `--pyargs`. See #4405 for context.

Reply:
Thank you for flagging that! I think I understand. The `testpaths` concept is overloaded to both refer to file system locations ("paths") and Python module names, and my change thus far requires `testpaths` to unconditionally resolve to file system locations.

I think a backwards-compatible solution would apply some additional intelligence to the way I'm processing `testpaths`. If an entry contains a wildcard, then treat it as a file system path and run it through `glob`; otherwise, accept it as a literal name that could either be path or a module (to be evaluated by later, existing logic).

Reply:
I believe I have this working as intended in the latest revision.

While this comment was associated with the `test_testpaths_ini` function, it's `test_collect_pyargs_with_testpaths` that needed to be reverted back to its original form. It specifically tests for the case you mention, and it was previously failing before I added the additional "does this path contain `glob`-style wildcards?" check.

Reply:
@jparise the real painful problem here is that "testpaths" is in fact only paths, not packages, pyargs was a tacked on hack to also look for tests in modules, its not soundly integrated

i'd love to eventually be able to havea consistent idea of collection roots and test id sources

Reply:
@RonnyPfannschmidt fortunately, I think we've arrived at a safe middle ground for this particular change, which only applies the `glob` expansion to paths when `--pyargs` _isn't_ specified, which should be unambiguous.